src/connectors/answers/connectAnswers.ts

```diff
@@ -0,0 +1,241 @@
+          },
+          true
+        );
+      },
+
+      render(renderOptions) {
+        const query = renderOptions.state.query;
+        if (!query) {
+          // renders nothing with empty query
+          lastResult = {};
+          isLoading = false;
+          renderFn(
+            {
+              ...this.getWidgetRenderState(renderOptions),
+              instantSearchInstance: renderOptions.instantSearchInstance,
+            },
+            false
+          );
+          return;
+        }
+
+        // render the loader
+        lastResult = {};
+        isLoading = true;
+        renderFn(
+          {
+            ...this.getWidgetRenderState(renderOptions),
+            instantSearchInstance: renderOptions.instantSearchInstance,
+          },
+          false
+        );
+
+        // call /answers API
+        runConcurrentSafePromise(
+          debouncedRefine(query, queryLanguages, {
+            nbHits,
+            attributesForPrediction,
+            // eslint-disable-next-line no-warning-comments
+            // FIXME: remove this x-algolia-agent once the engine accepts url encoded query params
+            queryParameters: {
+              'x-algolia-agent': 'answers-test',
+            },
+          })
+        )
+          .then(results => {
+            if (!results) {
+              // It's undefined when it's debounced.
+              return;
+            }
+
+            if (escapeHTML && results.hits.length > 0) {
+              results.hits = escapeHits(results.hits);
+            }
+            const initialEscaped = (results.hits as ReturnType<
+              typeof escapeHits
+            >).__escaped;
+
+            results.hits = addAbsolutePosition<typeof results.hits[0]>(
+              results.hits,
+              0,
+              nbHits
+            );
+
+            results.hits = addQueryID<typeof results.hits[0]>(
+              results.hits,
+              results.queryID
+            );
+
+            // Make sure the escaped tag stays, even after mapping over the hits.
+            // This prevents the hits from being double-escaped if there are multiple
+            // hits widgets mounted on the page.
+            (results.hits as ReturnType<
+              typeof escapeHits
+            >).__escaped = initialEscaped;
+
+            lastResult = results;
+            isLoading = false;
+            debouncedRender(
+              {
+                ...this.getWidgetRenderState(renderOptions),
+                instantSearchInstance: renderOptions.instantSearchInstance,
+              },
+              false
+            );
+          })
+          .catch(_error => {});
+      },
+
+      getRenderState(renderState, renderOptions) {
+        return {
+          ...renderState,
+          answers: this.getWidgetRenderState(renderOptions),
+        };
+      },
+
+      getWidgetRenderState() {
+        return {
+          hits: (lastResult && lastResult.hits) || [],
```

Comment:
where's refine?

Reply:
I didn't put it because I considered Answers widget like Hits widget. So the answers is refined by SearchBox, not by itself.